imapclient/client.go

```diff
@@ -796,7 +800,8 @@ func (c *Client) readResponseData(typ string) error {
 		}
 	}
 
-	switch typ {
+	// All response type are case insensitive
+	switch strings.ToUpper(typ) {
```

Comment:
Since `typ` is now upper-cased in the caller, I don't think we still need that one?

Reply:
Oh, just read your comment. Makes sense!